src/features/industry/industry.service.ts

```diff
@@ -11,15 +12,22 @@ export class IndustryService {
     ){}
     
     async findAll(): Promise<IndustryEntity[]> {
-        return this._industryRepository.find();
+        return await this._industryRepository.find();
     }
 
-    async findOne(id: number): Promise<IndustryEntity | undefined> {
-        return this._industryRepository.findOne(id);
+    async findOne(id: number): Promise<IndustryEntity> {
+        try{
+            return await this._industryRepository.findOneOrFail(id);
```

Comment:
You shouldn't have try-catch at handle the exceptions:

We need to specify the error message for the client:
```
let entity = await this._industryRepository.findOne(id);
if(!entity)
throw new NotFoundException('`ndsutry with provided id: ${id}` was not found);
```

Please refer to nest documentation to understand more about exception handlinging and to know which exception class to be used based on the error type: 
https://docs.nestjs.com/exception-filters